src/components/TokenSelect/index.test.tsx

```diff
@@ -35,11 +39,12 @@ describe('TokenSelect.tsx', () => {
         }
       )
 
-      await user.click(component.getByTestId('token-select'))
-      await expect(component.findByTestId('token-select-dialog')).resolves.toBeTruthy()
+      await user.click(component.getByRole('button'))
+      await waitFor(() => expect(component.getAllByText('Select a token').length).toBe(2))
```

Comment:
Can you find the header by role and then assert that the text content is 'Select a token', so it is less confusing (by just doubling the string)?

Reply:
This should not use `waitFor` - the change should be synchronous, and doing so prolongs the test duration.